schematic/store/synapse.py

```diff
@@ -81,31 +85,93 @@ class ManifestDownload(object):
-            # clear out all the existing manifests
-            if os.path.exists(temporary_manifest_storage):
-                shutil.rmtree(temporary_manifest_storage)
+            cleanup_temporary_storage(
+                temporary_manifest_storage, time_delta_seconds=3600
+            )
             # create a new directory to store manifest
             if not os.path.exists(temporary_manifest_storage):
                 os.mkdir(temporary_manifest_storage)
             # create temporary folders for storing manifests
-            download_location = create_temp_folder(temporary_manifest_storage)
+            download_location = create_temp_folder(
+                path=temporary_manifest_storage,
+                prefix=f"{self.manifest_id}-{time.time()}-",
+            )
         else:
-            download_location = CONFIG.manifest_folder
-        manifest_data = self.syn.get(
-            self.manifest_id,
-            downloadLocation=download_location,
-            ifcollision="overwrite.local",
+            if use_temporary_folder:
+                download_location = create_temp_folder(
+                    path=CONFIG.manifest_folder,
+                    prefix=f"{self.manifest_id}-{time.time()}-",
+                )
+            else:
+                download_location = CONFIG.manifest_folder
+
+        manifest_data = self.synapse_entity_tracker.get(
+            synapse_id=self.manifest_id,
+            syn=self.syn,
+            download_file=True,
+            retrieve_if_not_present=True,
+            download_location=download_location,
         )
+
+        # This is doing a rename of the downloaded file. The reason this is important
+        # is that if we are re-using a file that was previously downloaded, but the
+        # file had been renamed. The file downloaded from the Synapse client is just
+        # a direct copy of that renamed file. This code will set the name of the file
+        # to the original name that was used to download the file. Note: An MD5 checksum
+        # of the file will still be performed so if the file has changed, it will be
+        # downloaded again.
+        filename = manifest_data._file_handle.fileName
+        if filename != os.path.basename(manifest_data.path):
+            parent_folder = os.path.dirname(manifest_data.path)
+            manifest_original_name_and_path = os.path.join(parent_folder, filename)
+
+            self.syn.cache.remove(
+                file_handle_id=manifest_data.dataFileHandleId, path=manifest_data.path
+            )
+            os.rename(manifest_data.path, manifest_original_name_and_path)
+            manifest_data.path = manifest_original_name_and_path
+            self.syn.cache.add(
+                file_handle_id=manifest_data.dataFileHandleId,
+                path=manifest_original_name_and_path,
+                md5=manifest_data._file_handle.contentMd5,
+            )
+
         return manifest_data
```

Comment:
This is a potentially impactful change here @andrewelamb @linglp @milen-sage that could use a good look. Basically what this is solving:

(When NOT running on AWS)
1. When manifest were downloaded the `downloadAs` name, or the name of the actual file, could conflict with a file that was already in the location being downloaded to. This would overwrite the file and potentially cause the test, or whatever kick off that code to fail.

When running on either AWS or not

1. When downloading an entity where we already have a local copy of it (Think of a file that exists within the synapse cachemap) it will copy that file over to the requested location. The issue however is that the file that is copied retains whatever name it has been renamed to (Some tests do this, and code in schematic does this). Because of this I have the `os.rename` logic in place to set the name of the copied file back to the expected file name so that any subsequent code behaves as expected.

Reply:
I'm curious why renaming the file is a concern. Is that something schematic was doing?